src/BrandwatchReactAuth.js

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import jwtDecode from 'jwt-decode';
+import TokenStore from 'viziaauth';
+
+export default class BrandwatchReactAuth extends Component {
+  getChildContext() {
+    return {
+      brandwatchAuthLogout: this.handleLogout,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.store = new TokenStore(this.props.domain),
+    this.handleLogout = this.handleLogout.bind(this);
+
+    this.state = {
+      loggedIn: undefined,
+    };
+
+    props.onCreateStore(this.store);
+  }
+
+  componentWillMount() {
+    this.store.getToken({ aud: this.props.audience }).then((token) => {
+      if (!token) {
+        return window.location.href = this.store.loginUrl;
+      }
+
+      this.props.onAuthenticate({ token, profile: jwtDecode(token) });
```

Comment:
I'm a bit worried that calling `onAuthenticate` with the token could cause people to cache the token, what's the usecase for passing the token into `onAuthenticate`?